python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,298 @@
+            datasets=datasets,
+            preprocessor=preprocessor,
+            resume_from_checkpoint=resume_from_checkpoint,
+        )
+
+    @classmethod
+    def _create_trainer_loop_config(
+        cls,
+        lightning_module: pl.LightningModule,
+        lightning_module_config: Optional[Dict] = None,
+        lightning_trainer_config: Optional[Dict] = None,
+        lightning_trainer_fit_params: Optional[Dict] = None,
+        ddp_strategy_config: Optional[Dict] = None,
+        model_checkpoint_config: Optional[Dict] = None,
+        datasets_iter_config: Optional[Dict] = None,
+    ) -> Dict[str, Any]:
+
+        trainer_loop_config = {}
+        if not isclass(lightning_module):
+            raise ValueError(
+                "'lightning_module' must be a class, not a class instance."
+            )
+        if not issubclass(lightning_module, pl.LightningModule):
+            raise ValueError(
+                "'lightning_module' must be a subclass of "
+                "'pytorch_lightning.LightningModule'"
+            )
+        trainer_loop_config[LIGHTNING_MODULE_KEY] = lightning_module
+        trainer_loop_config[LIGHTNING_MODULE_CONFIG_KEY] = lightning_module_config
+        trainer_loop_config[LIGHTNING_TRAINER_CONFIG_KEY] = lightning_trainer_config
+        trainer_loop_config[LIGHTNING_TRAINER_FIT_PARAMS_KEY] = lightning_trainer_fit_params
+        trainer_loop_config[DDP_STRATEGY_CONFIG_KEY] = ddp_strategy_config
+        trainer_loop_config[MODEL_CHECKPOINT_CONFIG_KEY] = model_checkpoint_config
+        trainer_loop_config[RAY_DATASET_ITER_CONFIG_KEY] = datasets_iter_config
+
+        for key, config in trainer_loop_config.items():
+            if not config:
+                trainer_loop_config[key] = {}
+
+        return trainer_loop_config
+    
+    def _align_checkpoint_configs(self, model_checkpoint_config: Optional[Dict] = None):
+        if model_checkpoint_config is None:
+            # If no checkpoint config provided, always save the last checkpoint
+            model_checkpoint_config = {
+                "monitor": "epoch",
+                "mode": "max",
+                "save_top_k": 1
+            }
+            air_checkpoint_config = CheckpointConfig(
+                num_to_keep=1,
+            )
+        else:
+            air_checkpoint_config = CheckpointConfig(
+                num_to_keep=model_checkpoint_config.get("save_top_k", 1),
+                checkpoint_score_attribute=model_checkpoint_config.get("monitor", "epoch"),
+                checkpoint_score_order=model_checkpoint_config.get("mode", "max"),
+            )
+        return model_checkpoint_config, air_checkpoint_config
+
+
+def _prepare_dataloaders(dataloaders: Optional[DataLoader]) -> Optional[DataLoader]:
+    if dataloaders:
+        if isinstance(dataloaders, list):
+            for i, dataloader in enumerate(dataloaders):
+                dataloaders[i] = train.torch.prepare_data_loader(dataloader)
```

Comment:
ptl should handle this already?

Reply:
Just ran a test and you are right. I'll just remove these.